lightning/src/util/test_utils.rs

```diff
@@ -110,30 +114,61 @@ impl<'a> chain::Watch for TestChainMonitor<'a> {
 				&mut ::std::io::Cursor::new(&w.0)).unwrap() == update);
 
 		self.latest_monitor_update_id.lock().unwrap().insert(funding_txo.to_channel_id(), (funding_txo, update.update_id));
-		assert!(self.chain_monitor.update_channel(funding_txo, update).is_ok());
+		let update_res = self.chain_monitor.update_channel(funding_txo, update);
 		// At every point where we get a monitor update, we should be able to send a useful monitor
 		// to a watchtower and disk...
 		let monitors = self.chain_monitor.monitors.lock().unwrap();
-		let monitor = monitors.get(&funding_txo).unwrap();
-		w.0.clear();
-		monitor.write_for_disk(&mut w).unwrap();
-		let new_monitor = <(BlockHash, channelmonitor::ChannelMonitor<EnforcingChannelKeys>)>::read(
-				&mut ::std::io::Cursor::new(&w.0)).unwrap().1;
-		assert!(new_monitor == *monitor);
-		self.added_monitors.lock().unwrap().push((funding_txo, new_monitor));
+		match monitors.get(&funding_txo) {
+			Some(monitor) => {
+				w.0.clear();
+				monitor.write_for_disk(&mut w).unwrap();
+				let new_monitor = <(BlockHash, channelmonitor::ChannelMonitor<EnforcingChannelKeys>)>::read(
+					&mut ::std::io::Cursor::new(&w.0)).unwrap().1;
+				assert!(new_monitor == *monitor);
+				self.added_monitors.lock().unwrap().push((funding_txo, new_monitor));
+			},
+			_ => {}
```

Comment:
Document here that non-covered error is intentional and should be tested by caller  ?